ui/src/components/cake-day.tsx

```diff
@@ -0,0 +1,41 @@
+import { Component } from 'inferno';
+import moment from 'moment';
+import { i18n } from '../i18next';
+
+interface CakeDayProps {
+  creator_name: string;
+  creator_published: string;
+}
+
+export class CakeDay extends Component<CakeDayProps, any> {
+  render() {
+    const { creator_name, creator_published } = this.props;
+
+    return (
+      this.isCakeDay(creator_published) && (
+        <div
+          className="mr-lg-2 d-inline-block unselectable pointer mx-2"
```

Comment:
I would check to make sure these classes are the same as the other inline ones like the other badges, something tells me there are extras here.

Reply:
True. The additional class is `mx-2`.

Now we need this class when showing cake day icon in post listing, but we don't need it in comments as it's creating unnecessary additional gap.

Thus I created additional prop `is_post_creator` in `CakeDay` to add this class conditionally based on where `CakeDay` icon is showing.

Changed in 8fe8836bc263be11895fafa958409499c9e9fc9d.